iothub/device/tests/Authentication/ClientAuthenticationWithTokenRefreshTests.cs

```diff
@@ -240,6 +240,47 @@ public async Task ClientAuthenticationWithSakRefresh_WithModule_SharedAccessKeyC
             Assert.AreEqual(token1, token2.TokenValue);
         }
 
+        [TestMethod]
+        [ExpectedException(typeof(FormatException))]
+        public void ClientAuthenticationWithSakRefresh_InvalidKey_Throws()
+        {
+            var builder = new SharedAccessSignatureBuilder()
+            {
+                Key = TestSharedAccessKey + "123",
+            };
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(FormatException))]
+        public void ClientAuthenticationWithSakRefresh_InvalidKey1_Throws()
+        {
+            var TestSharedAccessKey_temp = TestSharedAccessKey.Replace('Y', '-');
+            var builder = new SharedAccessSignatureBuilder()
+            {
+                Key = TestSharedAccessKey_temp,
+            };
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(FormatException))]
+        public void ClientAuthenticationWithSakRefresh_NullKey_Throws()
+        {
+            var builder = new SharedAccessSignatureBuilder()
+            {
+                Key = null,
+            };
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(FormatException))]
+        public void ClientAuthenticationWithSakRefresh_EmptyKey_Throws()
+        {
+            var builder = new SharedAccessSignatureBuilder()
+            {
+                Key = "",
+            };
+        }
```

Comment:
Except with FluentAssertions, as above.

```suggestion
        [TestMethod]
        [ExpectedException(typeof(FormatException))]
        [DataRow(null)]
        [DataRow("")]
        public void ClientAuthenticationWithSakRefresh_InvalidKey_Throws(string key)
        {
            var builder = new SharedAccessSignatureBuilder
            {
                Key = key,
            };
        }
```